From cc48498e2c1708c11cb445d4aa38079507180052 Mon Sep 17 00:00:00 2001
From: David Clark <dclark@sierrawireless.com>
Date: Tue, 16 Jan 2018 13:44:14 -0800
Subject: [PATCH] Data Connection Service Stuck Pushes after WiFi/Cellular
 Disconnect

Updated AVC Service to reset authentication phase and session started flag
when a disconnected event is received.

Updated processing of updateState to not continue if its an unknown event.

Resolves: LE-9109
Change-Id: Ifec746143eb34e09aa6200222141e2c4fb227f13
---
 avcClient/avcClient.c | 1 +
 avcDaemon/avcServer.c | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/avcClient/avcClient.c b/avcClient/avcClient.c
index 0cb040d..cafa39c 100644
--- a/avcClient/avcClient.c
+++ b/avcClient/avcClient.c
@@ -304,6 +304,7 @@ static void ConnectionStateHandler
             // Call the callback.
             BearerEventCb(connected, contextPtr);
             DataConnected = false;
+            SessionStarted = false;
         }
     }
 }
diff --git a/avcDaemon/avcServer.c b/avcDaemon/avcServer.c
index 3cc8a40..b9bfbc6 100644
--- a/avcDaemon/avcServer.c
+++ b/avcDaemon/avcServer.c
@@ -1500,7 +1500,7 @@ static void ProcessUpdateStatus
             break;
 
         default:
-            LE_DEBUG("Unhandled updateStatus %s", AvcSessionStateToStr(data->updateStatus));
+            LE_WARN("Unhandled updateStatus %s", AvcSessionStateToStr(data->updateStatus));
             break;
     }
 
-- 
2.11.0

